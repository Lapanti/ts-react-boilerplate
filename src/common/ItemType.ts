enum ItemType {
  STORY = 'story',
  JOB = 'job',
  COMMENT = 'comment',
  POLL = 'poll',
  POLLOPT = 'pollopt',
}

export default ItemType;
