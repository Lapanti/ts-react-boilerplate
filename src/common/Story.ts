import ItemType from './ItemType';

export default interface Story {
  readonly by: string;
  readonly descendants: number;
  readonly id: number;
  readonly kids: number[];
  readonly score: number;
  readonly time: number;
  readonly title: string;
  readonly type: ItemType.STORY;
  readonly url: string;
}
