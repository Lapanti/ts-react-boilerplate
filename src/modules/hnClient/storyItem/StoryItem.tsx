import * as React from 'react';
import * as moment from 'moment';

import styled from '../../../theme/styled';
import Story from '../../../common/Story';

const StoryContainer = styled.div`
  align-items: flex-start;
  display: flex;
  font-size: 1.4rem;
  height: 100%;
  margin-top: 1rem;
`;

const InfoContainer = StoryContainer.extend`
  flex-direction: column;
  margin-left: 1rem;
  margin-top: 0;
`;

const StoryLink = styled.a`
  color: black;
  text-decoration: none;
`;

const MetaContainer = StoryContainer.extend`
  align-items: center;
  font-size: 1.2rem;
  margin-top: 0;
`;

export interface StoryItemProps {
  readonly story: Story;
  readonly index: number;
}

const StoryItem: React.StatelessComponent<StoryItemProps> = ({
  story,
  index,
}) => (
  <StoryContainer>
    {index + 1}.
    <InfoContainer>
      <StoryLink href={story.url} target="_blank">
        {story.title}
      </StoryLink>
      <MetaContainer>
        {story.score} points by {story.by} {moment.unix(story.time).fromNow()} |{' '}
        {story.descendants} comments
      </MetaContainer>
    </InfoContainer>
  </StoryContainer>
);

export default StoryItem;
