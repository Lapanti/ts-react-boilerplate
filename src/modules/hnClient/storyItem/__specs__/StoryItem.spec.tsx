import * as React from 'react';
import { shallow } from 'enzyme';
import createHistory from 'history/createBrowserHistory';
import * as moment from 'moment';

import StoryItem from '../StoryItem';
import Story from '../../../../common/Story';
import ItemType from '../../../../common/ItemType';

describe('<StoryItem />', () => {
  const hour = 3600;
  const year = 31556926;
  const story1: Story = {
    by: 'test',
    descendants: 10293,
    id: 1,
    kids: [],
    score: 1283,
    time: moment().unix() - hour,
    title: 'Test',
    type: ItemType.STORY,
    url: '',
  };

  const story2: Story = {
    by: 'test2',
    descendants: 10,
    id: 2,
    kids: [],
    score: 12,
    time: moment().unix() - year,
    title: 'Test 2',
    type: ItemType.STORY,
    url: '',
  };

  it('should render with correct props', () => {
    expect(shallow(<StoryItem story={story1} index={0} />)).toMatchSnapshot();
    expect(shallow(<StoryItem story={story2} index={1} />)).toMatchSnapshot();
  });
});
