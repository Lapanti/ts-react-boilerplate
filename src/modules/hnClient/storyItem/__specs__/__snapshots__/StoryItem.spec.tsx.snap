// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoryItem /> should render with correct props 1`] = `
<styled.div>
  1
  .
  <styled.div>
    <styled.a
      href=""
      target="_blank"
    >
      Test
    </styled.a>
    <styled.div>
      1283
       points by 
      test
       
      an hour ago
       |
       
      10293
       comments
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`<StoryItem /> should render with correct props 2`] = `
<styled.div>
  2
  .
  <styled.div>
    <styled.a
      href=""
      target="_blank"
    >
      Test 2
    </styled.a>
    <styled.div>
      12
       points by 
      test2
       
      a year ago
       |
       
      10
       comments
    </styled.div>
  </styled.div>
</styled.div>
`;
